stellar/contract/round/src/internal.rs

```diff
@@ -0,0 +1,802 @@
+        );
+        assert!(!round.is_completed, "Round is completed");
+
+        if round.use_whitelist {
+            let is_white_listed = is_white_listed(env, applicant.clone());
+            assert!(is_white_listed, "Applicant is not white listed");
+        }
+
+        let blacklist = is_black_listed(env, applicant.clone());
+        assert!(!blacklist, "Applicant is black listed");
+
+        let project_contract = read_project_contract(env);
+        let project_client = project_registry::Client::new(env, &project_contract);
+        let project = project_client.get_project_by_id(&project_id);
+        assert!(
+            project.is_some(),
+            "Project not found. Please register project first using project registry"
+        );
+
+        let existing_application = get_application(env, project_id);
+
+        assert!(existing_application.is_none(), "Application already exists");
+
+        let application_id = increment_application_number(env);
+        let review_note = String::from_str(env, "");
+        let application = ProjectApplication {
+            application_id,
+            project_id,
+            applicant,
+            status: ApplicationStatus::Pending,
+            submited_at: current_time,
+            review_note,
+            updated_at: None,
+        };
+
+        add_application(env, application.clone());
+        extend_instance(env);
+        log_project_application(env, application);
+
+        application_id
+    }
+
+    fn review_application(
+        env: &Env,
+        admin: Address,
+        application_id: u128,
+        status: ApplicationStatus,
+        note: Option<String>,
+    ) {
+        admin.require_auth();
+
+        let round = read_round_info(env);
+        let current_time = env.ledger().timestamp();
+
+        assert!(
+            round.application_start_time <= current_time
+                && current_time <= round.application_end_time,
+            "Application period is not live"
+        );
+        assert!(!round.is_completed, "Round is completed");
+
+        let application = get_application_by_id(env, application_id);
+
+        assert!(application.is_some(), "Application not found");
+
+        let mut updated_application = application.unwrap();
+        updated_application.status = status;
+
+        if note.is_some() {
+            let review_note = note.unwrap();
+            assert!(
+                review_note.len() <= 300,
+                "Review note must be less than 300 characters"
+            );
+            updated_application.review_note = review_note;
+        }
+
+        updated_application.updated_at = Some(current_time);
+
+        if updated_application.status == ApplicationStatus::Approved {
+            let approved_project = read_approved_projects(env);
+            assert!(
+                approved_project.len() < 10,
+                "Maximum approved projects {}",
+                10
+            );
```

Comment:
there's no limitation on maximum approved projects

Reply:
i think it's depend on the algorithm. let's make it parameterize